source/simulator/assembly.cc

```diff
@@ -1261,9 +1259,11 @@ namespace aspect
               const double density       = ((advection_field.is_temperature()) ? scratch.material_model_outputs.densities[q] : 1.0);
               const double conductivity  = ((advection_field.is_temperature()) ? scratch.material_model_outputs.thermal_conductivities[q] : 0.0);
               const double c_P           = ((advection_field.is_temperature()) ? scratch.material_model_outputs.specific_heat[q] : 1.0);
-              const double k_Delta_field = conductivity
-                                           * (scratch.old_field_laplacians[q] +
-                                              scratch.old_old_field_laplacians[q]) / 2;
+              const double k_Delta_field = ((advection_field.is_temperature()) ?
+                                            conductivity * (scratch.old_field_laplacians[q] +
+                                                            scratch.old_old_field_laplacians[q]) / 2
+                                            :
+                                            0.0);
```

Comment:
since you make sure that you're not touching the `conductivity` variable any more in this line, could you use `numbers::signaling_NaN<double>()` in the second branch of the initialization of `conductivity` above to be on the real safe side? I suspect that one could do the same for `density` and `c_P` as well.